math/vc_aos/include/algebra/math/impl/vc_aos_matrix.hpp

```diff
@@ -0,0 +1,127 @@
+  }
+
+  /// Operator getting one value of a const matrix
+  template <std::size_t ROWS, std::size_t COLS>
+  ALGEBRA_HOST_DEVICE static constexpr const value_type &element(
+      const matrix_type<ROWS, COLS> &m, std::size_t row, std::size_t col) {
+    return m[col][row];
+  }
+
+  /// Operator getting a block of a const matrix
+  template <std::size_t ROWS, std::size_t COLS, class input_matrix_t>
+  ALGEBRA_HOST_DEVICE static constexpr matrix_type<ROWS, COLS> block(
+      const input_matrix_t &m, std::size_t row, std::size_t col) {
+    return storage::block<ROWS, COLS>(m, row, col);
+  }
+
+  /// Operator setting a block with a vector matrix
+  template <std::size_t ROWS, std::size_t COLS, class input_matrix_t>
+  ALGEBRA_HOST_DEVICE static constexpr void set_block(
+      input_matrix_t &m, const matrix_type<ROWS, COLS> &b, std::size_t row,
+      std::size_t col) {
+    storage::set_block(m, b, row, col);
+  }
+
+  /// Operator setting a block with a vector
+  template <class input_matrix_t, class vector_t>
+  ALGEBRA_HOST_DEVICE static constexpr void set_block(input_matrix_t &m,
+                                                      const vector_t &b,
+                                                      std::size_t row,
+                                                      std::size_t col) {
+    storage::set_block(m, b, row, col);
+  }
+
+  // Create zero matrix
+  template <std::size_t ROWS, std::size_t COLS>
+  ALGEBRA_HOST_DEVICE static constexpr auto zero() {
+    return storage::zero<matrix_type<ROWS, COLS>>();
+  }
+
+  // Create identity matrix
+  template <size_t ROWS, size_t COLS>
+  ALGEBRA_HOST_DEVICE static constexpr auto identity() {
+    return storage::identity<matrix_type<ROWS, COLS>>();
+  }
+
+  // Set input matrix as zero matrix
+  template <std::size_t ROWS, std::size_t COLS>
+  ALGEBRA_HOST_DEVICE static constexpr void set_zero(
+      matrix_type<ROWS, COLS> &m) {
+    m = zero<ROWS, COLS>();
+  }
+
+  // Set input matrix as zero matrix
+  template <std::size_t ROWS, std::size_t COLS>
+  ALGEBRA_HOST_DEVICE static constexpr void set_identity(
+      matrix_type<ROWS, COLS> &m) {
+    m = identity<ROWS, COLS>();
+  }
+
+  /// Create transpose matrix
+  template <std::size_t ROWS, std::size_t COLS>
+  ALGEBRA_HOST_DEVICE static constexpr auto transpose(
+      const matrix_type<ROWS, COLS> &m) {
+    return storage::transpose(m);
+  }
+
+  /// Get determinant using a specific algorithm @tparam determinant_t
+  /*template <std::size_t N, typename determinant_t = determinant_t>
+  ALGEBRA_HOST_DEVICE static constexpr constexpr value_type determinant(
+      const matrix_type<N, N> &m) {
+
+    return determinant_t{}(m);
+  }
+
+  /// Get inverse using a specific algorithm @tparam inverse_t
+  template <std::size_t N, typename inverse_t = inverse_t>
+  ALGEBRA_HOST_DEVICE static constexpr constexpr matrix_type<N, N> inverse(
+      const matrix_type<N, N> &m) {
+
+    return inverse_t{}(m);
+  }*/
```

Comment:
Should we add disabled lines in this PR?

Reply:
I can remove them for now. They will be added in one of the other PRs